src/main/java/org/radarcns/producer/rest/SchemaRetriever.java

```diff
@@ -156,13 +152,14 @@ public void addSchemaMetadata(String topic, boolean ofValue, ParsedSchemaMetadat
                     .build();
 
             try (Response response = restClient.request(request)) {
+                String responseBody = response.body().string();
                 if (!response.isSuccessful()) {
                     throw new IOException("Cannot post schema to " + request.url()
                             + " (HTTP " + response.code() + ": " + response.message()
-                            + ") -> " + response.body().string());
+                            + ") -> " + responseBody);
                 }
-                JsonNode node = reader.readTree(response.body().byteStream());
-                int schemaId = node.get("id").asInt();
+                JSONObject node = new JSONObject(responseBody);
+                int schemaId = ((Number)node.get("id")).intValue();
```

Comment:
Could be just `node.getInt("id")`, `node.getInt("version")` and `node.getString("schema")` right?